src/coreComponents/codingUtilities/Utilities.hpp

```diff
@@ -118,6 +118,65 @@ T_VALUE softMapLookup( mapBase< T_KEY, T_VALUE, SORTED > const & theMap,
   return rvalue;
 }
 
+/**
+ * @brief Execute a user function on unique values in a range.
+ * @tparam ITER type of range iterator
+ * @tparam FUNC type of user function
+ * @tparam COMP type of comparison function
+ * @param first iterator to start of the range
+ * @param last iterator past the end of the range
+ * @param func the function to call, must be callable with value and size (as int)
+ *
+ * User function will be called once per consecutive group of equal values, as defined
+ * by @p comp, with a reference to one of the values from each group and the group size.
+ * If the range is (partially) ordered w.r.t. to a predicate compatible with @p comp,
+ * the function will be called once per unique value in the entire range.
+ *
+ * @note @p comp is an equality comparison, not a less-than type predicate used in sorting.
+ *       For a range sorted with some predicate P, one can use comp(x,y) = !(P(x,y) || P(y,x)),
+ *       but in most cases default value (std::equal_to<>) should be fine.
+ */
+template< typename ITER, typename FUNC, typename COMP = std::equal_to<> >
+void forEqualRanges( ITER first, ITER const last, FUNC && func, COMP && comp = {} )
```

Comment:
Not quite, the way you described it (if I read it correctly) the user function would receive pairs of **not-equal** elements. Whereas this function gives it sub-ranges of **equal** elements, so for input `[1,1,2,2,2,3,3,1]` the function would be called with `[1,1]`, `[2,2,2]`, `[3,3]`, `[1]` (passed via pairs of iterators into the input). I prefer to think of it as similar to [split_view](https://en.cppreference.com/w/cpp/ranges/split_view), only instead of splitting on (and removing) a particular value, it splits on adjacent-pair predicate `lhs != rhs` and preserves all values.

Reply:
But I definitely copy-pasted the documentation from `forUniqueValues`, which I will fix!

Reply:
Maybe `forUniformRanges` makes more sense that `forEqualRanges`?